README.md

```diff
@@ -1 +1,314 @@
+
+Signal Searcher (this repo) is an attempt to create an automated solution that will permit the display and discovery of issues relating to end-to-end Internet performance and aid in Internet performance transparency.  Once performance and measurement has been made transparent, network management can proceed from a firm footing based on open and auditable data. Signal searcher tries to help bootstrap this process by automatically mining the data.
+
+Let's break down Signal Searcher's mission for our data:
+
+> Signal Searcher attempts to find interesting and important signals in MLab's
+> Internet measurement data using computers instead of people.
+
+## Interesting
+
+A signal is interesting if:
+
+1. It has patterns that indicate overload
+2. It gets worse over time
+3. It changes differently than its cohort
+4. It is out of control
+5. It is surprising
+
+## Important
+
+A signal is important if:
+
+1. It corresponds to an attribute that directly affects a user's Internet
+   experience, e.g. latency, download throughput, and upload throughput.
+
+and either
+
+2. It has a large effect on part of the Internet OR
+3. It affects a large part of the Internet
+
+## Signal
+
+Each signal is a time-series of measured values. Signals may be (and almost
+certainly are) aggregates of time-series data. Let us imagine each value from a
+test as a single row in a table of measured values. Each value has lots of
+attributes for the test environment and for the measured values that were
+recorded. In data mining there is the concept of a [data cube](datacube): a
+hypercube where each dimension corresponds to a piece of the key specifying a
+data value. To create a signal, you must either specify an operation for every
+key dimension in the data cube.  That is, for every dimension you must
+"transform" (apply an operation to each data value that e.g. aligns timestamps,
+transforms IPs to ASNs, or buckets data), "roll up" (combine all data along the
+given dimension using some operation like "average" or "median"), or "drill
+down" (specify a particular value or range of values for that particular
+dimension). For metadata dimensions (dimensions that are neither part of the
+key or the value under test), there is another operator: "drop", which means to
+ignore that dimension entirely. Once every dimension has an operation
+specified, the result will be a table of values. If the operations are
+specified properly for our purposes, then the result will be a signal: a
+one-dimensional sequence of values.
+
+## MLab's Internet Measurement Data
+
+Our data is network measurement data, collected at the M-Lab measurement points
+across the Internet, and stored in [BigQuery](bigquery). The two main
+categories of data are measurements of performance and measurements of
+topology. Each test is associated with a litany of metadata regarding the
```

Comment:
A design doc for signal searcher doesn't seem like the right place for scheduling changes in how the pipeline works. No matter how much or how little metadata is in there, signal searcher will need to deal with it.